plugins/network-elements/netscaler/src/com/cloud/api/commands/DeployNetscalerVpxCmd.java

```diff
@@ -0,0 +1,149 @@
+    @Parameter(name = ApiConstants.TEMPLATE_ID, type = CommandType.UUID, entityType = TemplateResponse.class, required = true, description = "the ID of the template for the virtual machine")
+    private Long templateId;
+
+    @Parameter(name = ApiConstants.NETWORK_ID,
+            type = CommandType.UUID,
+            entityType = NetworkResponse.class, required=false,
+            description = "The network this ip address should be associated to.")
+    private Long networkId;
+    /////////////////////////////////////////////////////
+    /////////////////// Accessors ///////////////////////
+    /////////////////////////////////////////////////////
+
+
+
+    /////////////////////////////////////////////////////
+    /////////////// API Implementation///////////////////
+    /////////////////////////////////////////////////////
+
+    @Override
+    public void execute() throws ResourceUnavailableException, InsufficientCapacityException, ServerApiException, ConcurrentOperationException,
+        ResourceAllocationException {
+        try {
+            Map<String,Object> resp = _netsclarLbService.deployNetscalerServiceVm(this);
+           if (resp.size() > 0) {
+               SystemVmResponse response =  _responseGenerator.createSystemVmResponse((VirtualMachine)resp.get("vm"));
+               response.setGuestVlan((String)resp.get("guestvlan"));
+               response.setPublicVlan((List<String>)resp.get("publicvlan"));
+               response.setResponseName(getCommandName());
+               setResponseObject(response);
+           } else {
+               throw new ServerApiException(ApiErrorCode.INTERNAL_ERROR, "Fail to start system vm");
+           }
+
+        } catch (InvalidParameterValueException invalidParamExcp) {
+            throw new ServerApiException(ApiErrorCode.PARAM_ERROR, invalidParamExcp.getMessage());
+        } catch (CloudRuntimeException runtimeExcp) {
+            throw new ServerApiException(ApiErrorCode.INTERNAL_ERROR, runtimeExcp.getMessage());
+        }
+    }
+
+    public Long getServiceOfferingId() {
+        return serviceOfferingId;
+    }
+
+    public Long getTemplateId() {
+        return templateId;
+    }
+
+    @Override
+    public String getEventDescription() {
+        return "Adding a netscaler load balancer device";
+    }
+
+    @Override
+    public String getEventType() {
+        return EventTypes.EVENT_NETSCALER_VM_START;
+    }
+
+    @Override
+    public String getCommandName() {
+        return s_name;
+    }
+
+    public Long getZoneId() {
+        return zoneId;
+    }
+
+    @Override
+    public long getEntityOwnerId() {
+        return CallContext.current().getCallingAccount().getId();
+    }
+    public Account getAccount(){
+        return _accountService.getActiveAccountById(getEntityOwnerId());
+    }
+    public void getReservationContext() {
+        //ReservationContext context = new ReservationContextImpl(UUID.randomUUID().toString(), null , cmd.getAccount());
```

Comment:
@rajesh-battala : Right now, There is no usage of this function. So removal of this won't have any implication. And the return type should also be ReservationContext. I am correcting this code by returning the context, so that it can be used in future.